src/js/electron/menu/actions/searchActions.js

```diff
@@ -101,6 +101,26 @@ function buildSearchActions() {
         dispatch(submitSearch())
       }
     }),
+    jumpToTime: action({
+      name: "search-cell-menu-show-context",
+      label: "View in full context",
+      listener(dispatch, fieldData) {
+        let field = brim.field(fieldData)
+        let brimTime = brim.time(field.toDate())
+        if (field.type === "time") {
+          dispatch(tab.setFrom(brimTime.subtract(1, "minutes").toTs()))
+          dispatch(tab.setTo(brimTime.add(1, "minutes").toTs()))
+          SearchBar.clearSearchBar()
+          dispatch(SearchBar.changeSearchBarInput(""))
+          dispatch(submitSearch())
```

Comment:
Like you mentioned in the description, the setTimeout below won't be a reliable way to run code after the Viewer (table) has rendered. For that, it might make sense to have `submitSearch()` return a promise that resolves when the table has received all it's events. You could then write `dispatch(submitSearch()).then(() => /* Code that scrolls to log */)` 

As for scrolling to the proper log. For this, I think we should save the scroll position (maybe the x and y values) in the redux store. This would be in the `state/Viewer/reducer.js`. Then we could simply dispatch an action to set that state and then update the Viewer component to "react" to the change. Then we don't have to worry about timing the events just right.

Maybe a new file called src/js/flows/scrollToLog.js which could contain a redux "thunk" that looks something like this:

```js
export default const (log) => (dispatch, getState) => {
  const state = getState()
  const logs = Viewer.getLogs(state)
  const index = logs.findIndex(log2 => Log.is(log, log2))
  const rowHeight = 25 // pixels
  const scrollY = index * rowHeight
  
  dispatch(Viewer.setScroll(0, scrollY))
}
```

The this function in the searchActions might look like:
```js
/* set time and search bar ... */

dispatch(submitSearch()).then(() => {
  dispatch(scrollToLog(log))
})
```

Reply:
So everything is working. I'm not really sure about my approach though.

I've tried to solve this in `flows/submitSearch.js` initially. My approach was periodically reading the `Viewer.endState` variable to see if it was done fetching. However that required doing something periodically again and it wasn't working reliably. I couldn't figure out why.

So I changed around to the suggested Promise:

```javascript
/* .... */
    return new Promise(function(resolve) {
      switch (args.type) {
        case "analytics":
          dispatch(executeTableSearch(tabId, resolve, args))
          break
        case "zoom":
          dispatch(executeTableSearch(tabId, resolve, args))
          break
        default:
          dispatch(executeTableSearch(tabId, resolve, args))
          dispatch(executeHistogramSearch(tabId, args, prevArgs))
      }
    })
```

Let me know if there is a better way.